serving/samples/helloworld-nodejs/README.md

```diff
@@ -92,17 +92,13 @@ recreate the source files from this folder.
    # Create and change to the app directory.
    WORKDIR /usr/src/app
 
-   # Copy application dependency manifests to the container image.
-   # A wildcard is used to ensure both package.json AND package-lock.json are copied.
-   # Copying this separately prevents re-running npm install on every code change.
-   COPY package*.json ./
-
-   # Install production dependencies.
-   RUN npm install --only=production
-
    # Copy local code to the container image.
    COPY . .
 
+   # Install production dependencies but discard the node_modules 
+   # copied from your dev environment
+   RUN npm ci --only=production
```

Comment:
One problem is that now `npm ci` will be run anytime a source file changes. In the previous version, `npm install` was only ran if a change in `package*.json` happened.

Do you think there is any way to preserve the original optimization while also avoiding to copy local node_modules?

Reply:
there is not a simple way that i can tell to generate a correct installation without the user code.

the previous optimization could only every apply to a project without local install/build scripts so was really a false flag of comfort only applicable to hello world.

Reply:
you could consider changing it to
WORKDIR /usr/src/tmp
COPY package*.json ./
RUN npm ci --only-production --ignore-scripts
WORKDIR /usr/src/app
COPY . .
RUN cp -fr /usr/src/tmp/node_modules ./node_modules
RUN npm install --only=production --prefer-offline

this would cache all of the network IO but still run scripts when user code is in place

Reply:
@steren please let me know if you'ed prefer this solution ill update the pr